### PR TITLE
Shorten goods url and send to slack channel

### PR DESCRIPTION
**Pull request by thminhVN**
#### Status

READY

#### What's this PR do?

- [x] Shorten goods url
- [x] Show shorten url to slack
- [x] Show long url when go to shorten url
- [x] Disable button when user submit order form

### CODE REVIEW

**Comment by thminhVN** [comment]
@hieuphq @tieubao plz review it

File: web/controllers/order_controller.ex [hunk 1]
```diff
@@ -157,28 +191,14 @@ defmodule Delivr.OrderController do
...
-            ]
-          ],
+          fields: order_items,
           footer: "Delivr.to",
           footer_icon: "http://delivr.to/images/favicon.png",
           ts: to_timestamp(order.inserted_at)
         ]
       ]
     ]
+    IO.inspect payload
```
Comment (hieuphq) [inline web/controllers/order_controller.ex]: remove this statment

File: web/controllers/order_controller.ex [hunk 2]
```diff
@@ -148,6 +149,39 @@ defmodule Delivr.OrderController do
...
+        item_detail = [
+          title: "Link hàng",
+          value: "http://delivr.to/urls/" <> shorten_url(item.link)
+        ]
+      end
+    else
+      []
+    end
+
+    order_items = order_items ++ [
```
Comment (hieuphq) [inline web/controllers/order_controller.ex]: change order_items to other name, such as: field_datas

File: web/controllers/order_controller.ex [hunk 3]
```diff
@@ -323,5 +343,14 @@ defmodule Delivr.OrderController do
     end
   end
 
+  def shorten_url(long_url) do
+    changeset = Url.changeset(%Url{}, %{long_url: long_url})
+    case Repo.insert(changeset) do
+      {:ok, url} ->
+        Integer.to_string(url.id)
+      {:error, changeset} ->
```
Comment (hieuphq) [inline web/controllers/order_controller.ex]: add empty line before this statement

File: web/templates/url/index.html.eex [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        <%= link "Show", to: url_path(@conn, :show, url), class: "btn btn-default btn-xs" %>
+        <%= link "Edit", to: url_path(@conn, :edit, url), class: "btn btn-default btn-xs" %>
+        <%= link "Delete", to: url_path(@conn, :delete, url), method: :delete, data: [confirm: "Are you sure?"], class: "btn btn-danger btn-xs" %>
+      </td>
+    </tr>
+<% end %>
+  </tbody>
+</table>
+
+<%= link "New url", to: url_path(@conn, :new) %>
```
Comment (hieuphq) [inline web/templates/url/index.html.eex]: use 4 spaces instead of 2 spaces

File: web/templates/url/form.html.eex [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  <div class="form-group">
+    <%= label f, :long_url, class: "control-label" %>
+    <%= textarea f, :long_url, class: "form-control" %>
+    <%= error_tag f, :long_url %>
+  </div>
+
+  <div class="form-group">
+    <%= submit "Submit", class: "btn btn-primary" %>
+  </div>
+<% end %>
```
Comment (hieuphq) [inline web/templates/url/form.html.eex]: 4 spaces instead of 2 spaces

File: web/controllers/order_controller.ex [hunk 6]
```diff
@@ -157,22 +191,7 @@ defmodule Delivr.OrderController do
...
-              title: "Phone",
-              value: order.phone,
-              short: true
-            ],
-            [
-              title: "Email",
-              value: order.email
-            ]
-          ],
+          fields: order_items,
```
Comment (hieuphq) [inline web/controllers/order_controller.ex]: dm mai` >.<

**Comment by hieuphq** [comment]
👍 you can merge

**Pull request merged**